specification/deviceprovisioningservices/data-plane/Microsoft.Devices/stable/2021-10-01/service.json

```
@@ -0,0 +1,1465 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-10-01",
+    "title": "ProvisioningServiceClient",
+    "description": "API for service operations with the Azure IoT Hub Device Provisioning Service"
+  },
+  "host": "your-dps.azure-devices-provisioning.net",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/enrollments/{id}": {
+      "get": {
+        "tags": [
+          "GET"
+        ],
+        "summary": "Get a device enrollment record.",
+        "operationId": "IndividualEnrollment_Get",
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "id",
+            "in": "path",
+            "description": "Registration ID.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/IndividualEnrollment"
+            }
+          },
+          "default": {
+            "description": "Error response",
+            "schema": {
+              "$ref": "#/definitions/ProvisioningServiceErrorDetails"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PUT"
+        ],
+        "summary": "Create or update a device enrollment record.",
+        "operationId": "IndividualEnrollment_CreateOrUpdate",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "id",
+            "in": "path",
+            "description": "The registration ID is alphanumeric, lowercase, and may contain hyphens.",
+            "required": true,
+            "type": "string"
```

You should specify a `maxLength` and `pattern` on path parameters to restrict the length and allowed set of characters for user-specified names, e.g.
```suggestion
            "type": "string",
            "maxLength": 50,
            "pattern": "^[a-z0-9-]+$",
```

---

Good question! I hope that client SDKs do not validate this but I will confirm.

---

@mikekistler I have added the regex here, please confirm that SDKs do not validate and I will resolve this.

---

@leigharubin I looked into the code generated by autorest for a path parameter with a `pattern`, and I was surprised to see that the client does have logic to validate the pattern.  It’s not clear exactly when this logic is used, but the fact that it is there at all makes me second guess whether adding `pattern` here is a good idea.  It’s probably best to drop it for now.  Sorry …